### PR TITLE
[ROS2] grid_map_rviz_plugin port

### PR DESCRIPTION
**Pull request by Marwan99**
### Change
The [modified include](https://github.com/Marwan99/grid_map/tree/rviz-plugin-port/grid_map_rviz_plugin/include/grid_map_rviz_plugin/modified) files are not being used atm, the new [rviz_common::MessageFilterDisplay](https://github.com/ros2/rviz/blob/ros2/rviz_common/include/rviz_common/message_filter_display.hpp) seems (as far as I can tell) to be doing what was being done in the modified include files, is that okay?

### CODE REVIEW

**Comment by Marwan99** [comment]
CI expected to be failing till we have the CI updates from #253 merged.

**Comment by Marwan99** [comment]
First grid map in ROS2!
grid_map

**Comment by maximilianwulf** [comment]
> The modified include files are not being used atm, the new rviz_common::MessageFilterDisplay seems (as far as I can tell) to be doing what was being done in the modified include files, is that okay?

Yes, we had the modified include files because of the unconventional message definition. 

Thanks a lot.

File: grid_map_rviz_plugin/CMakeLists.txt [hunk 1]
```diff
@@ -71,44 +49,96 @@ set(SOURCE_FILES
...
+    list(APPEND AMENT_LINT_AUTO_EXCLUDE
+      ament_cmake_cpplint
+      ament_cmake_copyright
+    )
+    ament_lint_auto_find_test_dependencies()
+
+    # run cpplint without copyright filter
+    find_package(ament_cmake_cpplint)
+    ament_cpplint(
+      FILTERS -legal/copyright
```
Comment (maximilianwulf) [inline grid_map_rviz_plugin/CMakeLists.txt:127]: Is this a new feature of the MACRO?

File: grid_map_rviz_plugin/include/grid_map_rviz_plugin/GridMapDisplay.hpp [hunk 2]
```diff
@@ -6,74 +6,91 @@
  *  Institute: ETH Zurich, ANYbotics
  */
 
-#pragma once
+#ifndef GRID_MAP_RVIZ_PLUGIN__GRIDMAPDISPLAY_HPP_
+#define GRID_MAP_RVIZ_PLUGIN__GRIDMAPDISPLAY_HPP_
 
-#ifndef Q_MOC_RUN
+// #ifndef Q_MOC_RUN
```
Comment (maximilianwulf) [inline grid_map_rviz_plugin/include/grid_map_rviz_plugin/GridMapDisplay.hpp:12]: Can this just be deleted?

File: grid_map_rviz_plugin/include/grid_map_rviz_plugin/GridMapDisplay.hpp [hunk 3]
```diff
@@ -6,74 +6,91 @@
...
 #include <grid_map_ros/grid_map_ros.hpp>
-#include <grid_map_msgs/GridMap.h>
+#include <grid_map_msgs/msg/grid_map.hpp>
+
+#include <rviz_common/message_filter_display.hpp>
```
Comment (maximilianwulf) [inline grid_map_rviz_plugin/include/grid_map_rviz_plugin/GridMapDisplay.hpp:16]: Cool!

File: grid_map_rviz_plugin/include/grid_map_rviz_plugin/GridMapDisplay.hpp [hunk 4]
```diff
@@ -6,74 +6,91 @@
...
   void updateColorMode();
   void updateUseRainbow();
   void updateAutocomputeIntensityBounds();
   void updateVisualization();
 
- private:
+private:
   // Callback for incoming ROS messages
-  void processMessage(const grid_map_msgs::GridMap::ConstPtr& msg);
+  void processMessage(grid_map_msgs::msg::GridMap::ConstSharedPtr msg) override;
```
Comment (maximilianwulf) [inline grid_map_rviz_plugin/include/grid_map_rviz_plugin/GridMapDisplay.hpp:72]: Is the const intentionally removed?

File: grid_map_rviz_plugin/include/grid_map_rviz_plugin/GridMapVisual.hpp [hunk 5]
```diff
@@ -6,70 +6,86 @@
...
-namespace rviz {
+#include <grid_map_core/GridMap.hpp>
+#include <grid_map_msgs/msg/grid_map.hpp>
+#include <rviz_rendering/objects/billboard_line.hpp>
+
+#include <boost/shared_ptr.hpp>
+#include <string>
+#include <vector>
+
+// namespace Ogre
```
Comment (maximilianwulf) [inline grid_map_rviz_plugin/include/grid_map_rviz_plugin/GridMapVisual.hpp:27]: Delete candidate?

File: grid_map_rviz_plugin/include/grid_map_rviz_plugin/GridMapVisual.hpp [hunk 6]
```diff
@@ -6,70 +6,86 @@
...
 
   // ManualObject for mesh display.
-  Ogre::ManualObject* manualObject_;
+  Ogre::ManualObject * manualObject_;
   Ogre::MaterialPtr material_;
   std::string materialName_;
 
   // Lines for mesh.
-  boost::shared_ptr<rviz::BillboardLine> meshLines_;
+  boost::shared_ptr<rviz_rendering::BillboardLine> meshLines_;
```
Comment (maximilianwulf) [inline grid_map_rviz_plugin/include/grid_map_rviz_plugin/GridMapVisual.hpp:76]: Just out of curiosity, is there any compatibility to ROS1 with the RVIZ module?

File: grid_map_rviz_plugin/include/grid_map_rviz_plugin/modified/frame_manager.h [hunk 7]
```diff
@@ -27,31 +27,31 @@
...
+#include <tf2_ros/message_filter.h>
+
+#include <boost/thread/mutex.hpp>
 
+#include <map>
+#include <string>
+#include <algorithm>
 #include <type_traits>
 
 #ifndef Q_MOC_RUN
```
Comment (maximilianwulf) [inline grid_map_rviz_plugin/include/grid_map_rviz_plugin/modified/frame_manager.h:51]: Delete candidate?

**Review by maximilianwulf — COMMENTED** [review]
Hey looks great! Thanks. Can we just delete the modified header files?

File: grid_map_rviz_plugin/src/GridMapVisual.cpp [hunk 8]
```diff
@@ -6,33 +6,42 @@
  *  Institute: ETH Zurich, ANYbotics
  */
 
-#include <rviz/uniform_string_stream.h>
-#include <OGRE/OgreMaterialManager.h>
-#include <OGRE/OgreTextureManager.h>
-#include <OGRE/OgreTechnique.h>
+#ifndef Q_MOC_RUN
```
[the remark this answers is not in the document]
Comment (Marwan99) [reply grid_map_rviz_plugin/src/GridMapVisual.cpp]: Just added it while debugging, don't think they are needed.

File: grid_map_rviz_plugin/include/grid_map_rviz_plugin/GridMapDisplay.hpp (hunk 4 above)
Comment (Marwan99) [reply grid_map_rviz_plugin/include/grid_map_rviz_plugin/GridMapDisplay.hpp:70]: This is the [method](https://github.com/ros2/rviz/blob/d59072192205a92acd9bc84605420a815c885d00/rviz_common/include/rviz_common/message_filter_display.hpp#L214) being overridden. Also here is an [example](https://github.com/ros2/rviz/blob/d59072192205a92acd9bc84605420a815c885d00/rviz_default_plugins/src/rviz_default_plugins/displays/camera/camera_display.cpp#L568) of it being used.

I tried to stick to the same parameter type as in the virtual function above and assumed `::ConstSharedPtr` has the const part already. Having said that, I just tried adding const back and it is also building fine, not sure if there is a need to have it back though.

**Comment by Marwan99** [comment]
@SteveMacenski please review.

**Review by SteveMacenski — APPROVED** [review]

File: grid_map_rviz_plugin/include/grid_map_rviz_plugin/GridMapVisual.hpp (hunk 6 above)
Comment (maximilianwulf) [reply grid_map_rviz_plugin/include/grid_map_rviz_plugin/GridMapVisual.hpp:68]: Okay, thanks.

File: grid_map_rviz_plugin/include/grid_map_rviz_plugin/modified/frame_manager.h [hunk 9]
```diff
@@ -1,333 +0,0 @@
-/*
```
Comment (maximilianwulf) [inline grid_map_rviz_plugin/include/grid_map_rviz_plugin/modified/frame_manager.h:1]: Nice.

**Review by maximilianwulf — APPROVED** [review]
LGTM, thanks agian.

**Pull request merged**